aws/logs_monitoring/lambda_function.py

```diff
@@ -277,8 +281,14 @@ def s3_handler(event, context, metadata):
             )
             yield structured_line
     else:
+        # Check if using multiline log regex pattern
+        if DD_MULTILINE_LOG_REGEX_PATTERN:
+            split_data = re.compile("(?<!^)\s+(?=%s)(?!.\s)" % (DD_MULTILINE_LOG_REGEX_PATTERN,)).split(data)
```

Comment:
what happens when there is no match ? i.e. there is two different kind of inputs, a pattern separated one and a line separated one ?

Reply:
I see your point, in the case where there is both types of s3 logs and the format varies, the line separated kind would not get split out into separate log lines.  It assumes each log line, single or multi, begins with the same pattern at the start of the line ie `(?<!^)\s+(?=<insert_pattern>)`

Open to suggestions on how to approach this, I pushed up a somewhat hacky attempt to solve this (i think it is a solution at least ) which checks, when a multiline_regex env var is supplied, if there are _any_ matches for that pattern ,and if there aren't then assumes it is line separated and calls `.splitlines()` as usual. My understanding is that every `event` 
 passed into `s3_handler()` would contain only 1 type, either pattern or line separated, but if that is not the case please let me know.

Also open to scrapping this we feel like it opens up a huge rabbit hole